src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrPatchHandler.cs

```diff
@@ -205,13 +208,54 @@ .. mapping.Exclude.Select(GetExclusionRule),
         return patches;
     }
 
+    /// <summary>
+    /// Applies a set of patches by applying all patches, preserving conflicts in the index.
+    /// In case of failures, collects all conflicted files into a single exception.
+    /// </summary>
+    public async Task ApplyPatches(
+        IEnumerable<VmrIngestionPatch> patches,
+        NativePath targetDirectory,
+        bool removePatchAfter,
+        bool keepConflicts,
+        bool reverseApply = false,
+        CancellationToken cancellationToken = default)
+    {
+        List<UnixPath> conflicts = [];
+
+        foreach (var patch in patches)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                await ApplyPatch(
+                    patch,
+                    targetDirectory,
+                    removePatchAfter: removePatchAfter,
+                    keepConflicts: keepConflicts,
+                    reverseApply: reverseApply,
+                    cancellationToken);
+            }
+            catch (PatchApplicationLeftConflictsException e)
+            {
+                conflicts.AddRange(e.ConflictedFiles);
+            }
+        }
+
+        if (conflicts.Count > 0)
+        {
+            throw new PatchApplicationLeftConflictsException([..conflicts.Distinct()], targetDirectory);
```

Comment:
who ends up catching this? I imagine it should be pull request updater, or maybe we just add the comment collector here?

Reply:
@dkurepa it will be the `PullRequestUpdater` or it's also done by the `Back`/`ForwardFlowOperation`